src/Estimators/src/UkfState.cpp

```diff
@@ -0,0 +1,543 @@
+            {
+                FTMap[key] = currentState.segment(stateVariableHandler.getVariable(key).offset,
+                                                  stateVariableHandler.getVariable(key).size);
+            }
+
+            for (int idx = 0; idx < subModelList[subModelIdx].getNrOfExternalContact(); idx++)
+            {
+                extContactMap[subModelList[subModelIdx].getExternalContact(idx)]
+                    = currentState
+                          .segment(stateVariableHandler
+                                       .getVariable(
+                                           subModelList[subModelIdx].getExternalContact(idx))
+                                       .offset,
+                                   stateVariableHandler
+                                       .getVariable(
+                                           subModelList[subModelIdx].getExternalContact(idx))
+                                       .size);
+            }
+        }
+    }
+
+    bool updateState()
+    {
+        // Update kindyn full model
+        kinDynFullModel->setRobotState(ukfInput.robotBasePose.transform(),
+                                       ukfInput.robotJointPositions,
+                                       iDynTree::make_span(ukfInput.robotBaseVelocity.data(),
+                                                           manif::SE3d::Tangent::DoF),
+                                       jointVelocityState,
+                                       gravity);
+
+        // compute joint acceleration per each sub-model
+        for (int subModelIdx = 0; subModelIdx < subModelList.size(); subModelIdx++)
+        {
+            // Update the kindyn wrapper object of the submodel
+            kinDynWrapperList[subModelIdx]->updateState(ukfInput.robotBaseAcceleration,
+                                                        jointAccelerationState,
+                                                        UpdateMode::RobotDynamicsOnly);
+
+            if (subModelList[subModelIdx].getModel().getNrOfDOFs() > 0)
+            {
+                totalTorqueFromContacts[subModelIdx].setZero();
+
+                // Contribution of FT measurements
+                for (auto& [key, value] : subModelList[subModelIdx].getFTList())
+                {
+                    wrench = static_cast<int>(value.forceDirection) * FTMap[key].array();
+
+                    torqueFromContact[subModelIdx]
+                        = kinDynWrapperList[subModelIdx]
+                              ->getFTJacobian(key)
+                              .block(0, 6, 6, subModelList[subModelIdx].getModel().getNrOfDOFs())
+                              .transpose()
+                          * wrench;
+
+                    totalTorqueFromContacts[subModelIdx]
+                        = totalTorqueFromContacts[subModelIdx].array()
+                          + torqueFromContact[subModelIdx].array();
```

Comment:
It shouldn't be required to covert in array. If not required can you remove it also in other parts?